File: android/java/StorylyDemo/app/src/main/java/com/appsamurai/storylydemo/use_cases/HideActivity.java
package com.appsamurai.storylydemo.use_cases;

import android.os.Bundle;
import android.os.Handler;
import android.widget.LinearLayout;

import androidx.appcompat.app.AppCompatActivity;

import com.appsamurai.storyly.Story;
import com.appsamurai.storyly.StoryComponent;
import com.appsamurai.storyly.StoryGroup;
import com.appsamurai.storyly.StorylyInit;
import com.appsamurai.storyly.StorylyListener;
import com.appsamurai.storyly.StorylyView;
import com.appsamurai.storyly.analytics.StorylyEvent;
import com.appsamurai.storylydemo.R;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public class HideActivity extends AppCompatActivity {
    StorylyView storylyView;
    LinearLayout storylyViewHolder;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_hide);
        getSupportActionBar().hide();

        storylyView = findViewById(R.id.storyly_view);
        storylyViewHolder = findViewById(R.id.storyly_view_holder);

        String STORYLY_INSTANCE_TOKEN = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhY2NfaWQiOjc2MCwiYXBwX2lkIjo0MDUsImluc19pZCI6NDA0fQ.1AkqOy_lsiownTBNhVOUKc91uc9fDcAxfQZtpm3nj40";

        storylyView.setStorylyInit(new StorylyInit(STORYLY_INSTANCE_TOKEN));

        storylyView.setStorylyListener(new StorylyListener() {
            boolean storylyLoaded = false;

            @Override
            public void storylyLoaded(@NotNull StorylyView storylyView, @NotNull List<StoryGroup> list) {
                if (list.size() > 0) {
                    storylyLoaded = true;
                }
            }

            @Override
            public void storylyLoadFailed(@NotNull StorylyView storylyView, @NotNull String s) {
                // if cached before not hide
                if (!storylyLoaded) {
                    new Handler(getMainLooper()).postDelayed(new Runnable() {
                        @Override
                        public void run() {
                            storylyViewHolder.removeView(HideActivity.this.storylyView);
                        }
                    }, 400);
                }
            }

            @Override
            public void storylyActionClicked(@NotNull StorylyView storylyView, @NotNull Story story) { }

            @Override
            public void storylyStoryShown(@NotNull StorylyView storylyView) { }

            @Override
            public void storylyStoryDismissed(@NotNull StorylyView storylyView) { }

            @Override
            public void storylyUserInteracted(@NotNull StorylyView storylyView, @NotNull StoryGroup storyGroup, @NotNull Story story, @NotNull StoryComponent storyComponent) { }

            @Override
            public void storylyEvent(@NotNull StorylyView storylyView, @NotNull StorylyEvent storylyEvent, @Nullable StoryGroup storyGroup, @Nullable Story story, @Nullable StoryComponent storyComponent) { }
        });
    }
}

File: android/kotlin/StorylyDemo/app/src/main/java/com/appsamurai/storylydemo/use_cases/HideActivity.kt
package com.appsamurai.storylydemo.use_cases

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.appsamurai.storyly.*
import com.appsamurai.storylydemo.STORYLY_INSTANCE_TOKEN
import com.appsamurai.storylydemo.databinding.ActivityHideBinding


class HideActivity: AppCompatActivity() {
    private lateinit var binding: ActivityHideBinding


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityHideBinding.inflate(layoutInflater)
        setContentView(binding.root)
        supportActionBar?.hide()

        binding.storylyView.storylyInit = StorylyInit(STORYLY_INSTANCE_TOKEN)

        binding.storylyView.storylyListener = object : StorylyListener {
            var storylyLoaded = false

            override fun storylyLoaded(storylyView: StorylyView, storyGroupList: List<StoryGroup>, dataSource: StorylyDataSource) {
                if (storyGroupList.isNotEmpty()) {
                    storylyLoaded = true
                }
            }

            override fun storylyLoadFailed(storylyView: StorylyView, errorMessage: String) {
                // if cached before not hide
                if (!storylyLoaded) {
                    storylyView.visibility = View.GONE;
                }
            }
        }
    }
}